WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/usecases/ServiceMapper.kt

```diff
@@ -0,0 +1,63 @@
+import org.wordpress.android.ui.stats.refresh.usecases.ServiceMapper.Service.GOOGLE_PLUS
+import org.wordpress.android.ui.stats.refresh.usecases.ServiceMapper.Service.LINKED_IN
+import org.wordpress.android.ui.stats.refresh.usecases.ServiceMapper.Service.PATH
+import org.wordpress.android.ui.stats.refresh.usecases.ServiceMapper.Service.TUMBLR
+import org.wordpress.android.ui.stats.refresh.usecases.ServiceMapper.Service.TWITTER
+import org.wordpress.android.viewmodel.ResourceProvider
+import javax.inject.Inject
+
+private const val FACEBOOK_ICON = "https://secure.gravatar.com/blavatar/2343ec78a04c6ea9d80806345d31fd78?s="
+private const val TWITTER_ICON = "https://secure.gravatar.com/blavatar/7905d1c4e12c54933a44d19fcd5f9356?s="
+private const val TUMBLR_ICON = "https://secure.gravatar.com/blavatar/84314f01e87cb656ba5f382d22d85134?s="
+private const val GOOGLE_PLUS_ICON = "https://secure.gravatar.com/blavatar/4a4788c1dfc396b1f86355b274cc26b3?s="
+private const val LINKED_IN_ICON = "https://secure.gravatar.com/blavatar/f54db463750940e0e7f7630fe327845e?s="
+private const val PATH_ICON = "https://secure.gravatar.com/blavatar/3a03c8ce5bf1271fb3760bb6e79b02c1?s="
+
+class ServiceMapper
+@Inject constructor(private val resourceProvider: ResourceProvider) {
+    fun map(services: List<org.wordpress.android.fluxc.model.stats.PublicizeModel.Service>): List<BlockListItem.Item> {
+        val dimension = resourceProvider.getDimensionPixelSize(R.dimen.avatar_sz_small)
+        return services.mapIndexed { index, service ->
+            val mappedService = getService(service.name)
+            Item(
+                    iconUrl = mappedService?.iconUrl?.let { it + dimension },
+                    text = if (mappedService?.nameResource == null) service.name else null,
+                    textResource = mappedService?.nameResource,
+                    value = service.followers.toFormattedString(),
+                    showDivider = index < services.size - 1
+            )
+        }
+    }
+
+    enum class Service(
+        val iconUrl: String,
+        @StringRes val nameResource: Int
+    ) {
+        FACEBOOK(FACEBOOK_ICON, nameResource = R.string.stats_insights_facebook),
+        TWITTER(TWITTER_ICON, nameResource = R.string.stats_insights_twitter),
+        TUMBLR(TUMBLR_ICON, nameResource = R.string.stats_insights_tumblr),
+        GOOGLE_PLUS(GOOGLE_PLUS_ICON, nameResource = R.string.stats_insights_google_plus),
+        LINKED_IN(LINKED_IN_ICON, nameResource = R.string.stats_insights_linkedin),
+        PATH(PATH_ICON, nameResource = R.string.stats_insights_path)
+    }
+
+    private fun getService(service: String): Service? {
+        return when (service) {
+            "facebook" -> FACEBOOK
+            "twitter" -> TWITTER
+            "google_plus" -> GOOGLE_PLUS
+            "tumblr" -> TUMBLR
+            "linkedin" -> LINKED_IN
+            "path" -> PATH
+            else -> null
```

Comment:
The original behaviour is that when we have a service that's not one of the implicitly supported ones, we show the `label` capitalized without an icon (but still with data). I don't think this would ever happen but it was the previous behaviour of the fragment.